File: codewars/7kyu/array-leaders-(array-series-6)/dduck-array-leaders-(array-series-6).py
# https://www.codewars.com/kata/5a905c2157c562994900009d

from unittest import TestCase


def minimum_steps(numbers, value):
    sum = 0
    for i, x in enumerate(sorted(numbers)):
        sum += x
        if sum >= value:
            return i
    return -1


TestCase().assertEqual(minimum_steps([4, 6, 3], 7), 1)
TestCase().assertEqual(minimum_steps([10, 9, 9, 8], 17), 1)
TestCase().assertEqual(minimum_steps([8, 9, 10, 4, 2], 23), 3)
TestCase().assertEqual(minimum_steps([19, 98, 69, 28, 75, 45, 17, 98, 67], 464), 8)
TestCase().assertEqual(minimum_steps([4, 6, 3], 2), 0)

File: codewars/7kyu/array-leaders-(array-series-6)/dduck-array-leaders-(array-series-6).js
// https://www.codewars.com/kata/5a91a7c5fd8c061367000002/

const minimumSteps = (numbers, v) => numbers.sort((a, b) => a - b)
  .reduce((a, c, i, arr) => {
    const sum = a + c;
    if (sum >= v) {
      arr.splice(1);
      return i;
    }
    return a + c;
  }, 0);

console.log(minimumSteps([4, 6, 3], 7), 1);
console.log(minimumSteps([10, 9, 9, 8], 17), 1);
console.log(minimumSteps([8, 9, 10, 4, 2], 23), 3);
console.log(minimumSteps([19, 98, 69, 28, 75, 45, 17, 98, 67], 464), 8);
console.log(minimumSteps([4, 6, 3], 2), 0);
